### ISSUE TITLE
Get an error run `scripts/test` in a newly cloned `uvicorn`

### ISSUE THREAD

**Issue by Kludex** [Kludex/uvicorn#1763]
### Discussed in https://github.com/encode/uvicorn/discussions/1755

<div type='discussions-op-text'>

<sup>Originally posted by **ys-wu** November  2, 2022</sup>
I clone my fork: 
```
git clone https://github.com/MY-USERNAME/uvicorn
```

Then simply install and test:
```
cd uvicorn
scripts/install
scripts/test
```

And get the error:
```
+ '[' -z ']'
+ scripts/check
+ ./scripts/sync-version
+ venv/bin/black --check --diff --target-version=py37 uvicorn tests
All done! ✨ 🍰 ✨
63 files would be left unchanged.
+ venv/bin/flake8 uvicorn tests
+ venv/bin/mypy --show-error-codes
Success: no issues found in 53 source files
+ venv/bin/isort --check --diff --project=uvicorn uvicorn tests
+ venv/bin/python -m tools.cli_usage --check
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/wuyusheng/playground/uvicorn/tools/cli_usage.py", line 69, in <module>
    rv |= _generate_cli_usage(path, check=args.check)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/wuyusheng/playground/uvicorn/tools/cli_usage.py", line 47, in _generate_cli_usage
    usage_lines = _get_usage_lines()
                  ^^^^^^^^^^^^^^^^^^
  File "/Users/wuyusheng/playground/uvicorn/tools/cli_usage.py", line 13, in _get_usage_lines
    res = subprocess.run(["uvicorn", "--help"], stdout=subprocess.PIPE)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/wuyusheng/.pyenv/versions/3.11.0/lib/python3.11/subprocess.py", line 546, in run
    with Popen(*popenargs, **kwargs) as process:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/wuyusheng/.pyenv/versions/3.11.0/lib/python3.11/subprocess.py", line 1022, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/Users/wuyusheng/.pyenv/versions/3.11.0/lib/python3.11/subprocess.py", line 1899, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'uvicorn'
```

My environment:
```
MacBook Pro 2017
macOS Monterey Version 12.6 (21G115)
Python 3.11.0
```

</div>

**Comment by Ou7law007** [comment]
I'm also having problems with python3.11 with uvicorn[standard] installation, I don't know if I should open an issue though...

```
httptools/parser/parser.c:212:12: fatal error: longintrepr.h: No such file or directory
        212 |   #include "longintrepr.h"
            |            ^~~~~~~~~~~~~~~
      compilation terminated.
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> httptools
```

**Comment by Kludex** [comment]
> I'm also having problems with python3.11 with uvicorn[standard] installation, I don't know if I should open an issue though...
> 
> ```
> httptools/parser/parser.c:212:12: fatal error: longintrepr.h: No such file or directory
>         212 |   #include "longintrepr.h"
>             |            ^~~~~~~~~~~~~~~
>       compilation terminated.
>       error: command '/usr/bin/gcc' failed with exit code 1
>       [end of output]
>   
>   note: This error originates from a subprocess, and is likely not a problem with pip.
> error: legacy-install-failure
> 
> × Encountered error while trying to install package.
> ╰─> httptools
> ```

That looks like an issue to `httptools`.

I'll hide those two comments because they are spam, jfyk.

**Comment by Kludex** [comment]
I understand the problem now.

The problem is that we don't have `uvicorn` in the system path, so the OS can't find `uvicorn`. This was not a problem before because when you are developing with `uvicorn`, I guess it's assumed that you are able to run `uvicorn` via CLI.

I tried to use `sys.executable` on `tools/cli_usage.py`, but then the output message changes... Maybe a regex replace can be added there? I'll wait for a cleaner solution to appear. :sweat_smile:

**Comment by cdeler** [comment]
Let me try to do that. I haven't contributed for a while and forgot everything 🥲

**Comment by cdeler** [comment]
https://github.com/encode/uvicorn/pull/1823 here is my attempt to implement it

**Comment by Kludex** [comment]
> #1823 here is my attempt to implement it

Closed by #1823 :pray:
